src/internal.c

```diff
@@ -6305,6 +6333,44 @@ int WP11_Object_SetSecretKey(WP11_Object* object, unsigned char** data,
     return ret;
 }
 
+#ifdef WOLFPKCS11_NSS
+int WP11_Object_SetTrust(WP11_Object* object, unsigned char** data,
+                         CK_ULONG* len)
+{
+    int ret = 0;
+    WP11_Trust* trust;
+
+    if (object->onToken)
+        WP11_Lock_LockRW(object->lock);
+
+    trust = &object->data.trust;
+
+    if (ret == 0 && data[0] == NULL && len[0] != WC_SHA_DIGEST_SIZE)
+        ret = BAD_FUNC_ARG;
+    if (ret == 0 && data[0] != NULL)
+        XMEMCPY(trust->sha1Hash, data[0], WC_SHA_DIGEST_SIZE);
+    if (ret == 0 && data[1] == NULL && len[1] != WC_MD5_DIGEST_SIZE)
+        ret = BAD_FUNC_ARG;
+    if (ret == 0 && data[1] != NULL)
+        XMEMCPY(trust->md5Hash, data[1], WC_MD5_DIGEST_SIZE);
+    if (ret == 0 && data[2] != NULL)
+        trust->serverAuth = *(CK_ULONG*)data[2];
+    if (ret == 0 && data[3] != NULL)
+        trust->clientAuth = *(CK_ULONG*)data[3];
+    if (ret == 0 && data[4] != NULL)
+        trust->emailProtection = *(CK_ULONG*)data[4];
+    if (ret == 0 && data[5] != NULL)
+        trust->codeSigning = *(CK_ULONG*)data[5];
+    if (ret == 0 && data[6] != NULL)
+        trust->stepUpApproved = *(CK_BBOOL*)data[6];
+
+    if (object->onToken)
+        WP11_Lock_UnlockRW(object->lock);
+
+    return ret;
+}
```

Comment:
The return cascade is not necessary if it is only used for input validation.

```suggestion
int WP11_Object_SetTrust(WP11_Object* object, unsigned char** data,
                         CK_ULONG* len)
{
    WP11_Trust* trust;

    if (data[0] == NULL && len[0] != WC_SHA_DIGEST_SIZE)
        return BAD_FUNC_ARG;
    if (data[1] == NULL && len[1] != WC_MD5_DIGEST_SIZE)
        return BAD_FUNC_ARG;

    if (object->onToken)
        WP11_Lock_LockRW(object->lock);

    trust = &object->data.trust;

    if (data[0] != NULL)
        XMEMCPY(trust->sha1Hash, data[0], WC_SHA_DIGEST_SIZE);
    if (data[1] != NULL)
        XMEMCPY(trust->md5Hash, data[1], WC_MD5_DIGEST_SIZE);
    if (data[2] != NULL)
        trust->serverAuth = *(CK_ULONG*)data[2];
    if (data[3] != NULL)
        trust->clientAuth = *(CK_ULONG*)data[3];
    if (data[4] != NULL)
        trust->emailProtection = *(CK_ULONG*)data[4];
    if (data[5] != NULL)
        trust->codeSigning = *(CK_ULONG*)data[5];
    if (data[6] != NULL)
        trust->stepUpApproved = *(CK_BBOOL*)data[6];

    if (object->onToken)
        WP11_Lock_UnlockRW(object->lock);

    return ret;
}
```